### PR TITLE
add clientOption for allowScroll

### PR DESCRIPTION
**Pull request by justinchou-google**
[b/238669349](https://b.corp.google.com/issues/238669349)

Adding client option to allow scrolling behind a subscription or contribution dialog, called `allowScroll`.

### CODE REVIEW

File: src/runtime/client-config-manager.js [hunk 1]
```diff
@@ -112,6 +112,15 @@ export class ClientConfigManager {
     return this.clientOptions_.theme || ClientTheme.LIGHT;
   }
 
+  /**
+   * Returns whether scrolling on main page should be allowed when
+   * subscription or contribution dialog is displayed.
+   * @return {boolean}
+   */
+  shouldAllowScroll() {
```
Comment (justinchou-google) [inline src/runtime/client-config-manager.js:120]: may be worth converting this to `shouldDisableScroll` or (some other negation) to match the config `shouldDisableBodyScrolling`

**Comment by justinchou-google** [comment]
Example of `allowScroll` for subscriptions prompt (note the scroll bar on the right):
Screen Shot 2022-08-17 at 5 07 12 PM

File: src/runtime/offers-flow.js [hunk 2]
```diff
@@ -261,11 +261,11 @@ export class OffersFlow {
    * @param {!../model/client-config.ClientConfig} clientConfig
```
Comment (oyj9109) [inline src/runtime/offers-flow.js:261]: nit: update description for the new param.

File: src/runtime/contributions-flow.js [hunk 3]
```diff
@@ -147,7 +147,7 @@ export class ContributionsFlow {
           return this.dialogManager_.openView(
             this.activityIframeView_,
             /* hidden */ false,
-            this.getDialogConfig_(clientConfig)
+            this.getDialogConfig_(clientConfig, this.clientConfigManager_)
```
Comment (oyj9109) [inline src/runtime/contributions-flow.js:150]: How about calling `shouldAllowScroll` here and passing a boolean in instead?

File: src/runtime/offers-flow.js (hunk 2 above)
Comment (justinchou-google) [reply src/runtime/offers-flow.js:261]: Missed this, good catch!

File: src/runtime/client-config-manager.js (hunk 1 above)
Comment (justinchou-google) [reply src/runtime/client-config-manager.js:120]: Will leave as `shouldAllowScroll`.

**Review by oyj9109 — APPROVED** [review]
LGTM. Thanks for leaving detailed notes and screencasts. 
I know there's a discussion on whether to apply this to other prompts, but I think we can follow up in a separate PR if needed.

**Review by jpettitt — APPROVED** [review]

**Pull request merged**